#! /usr/bin/env python

PACKAGE='wj_716N_lidar'
from dynamic_reconfigure.parameter_generator_catkin import *
#from math import pi

#from driver_base.msg import SensorLevels

gen = ParameterGenerator()
#       Name              Type      Reconfiguration level             Description                                      Default    Min       Max
gen.add("min_ang",        double_t, 0, "The angle of the first range measurement [rad].",                               -3.14,    -3.14,    3.14)
gen.add("max_ang",        double_t, 0, "The angle of the last range measurement [rad].",                                3.14,     -3.14,    3.14)
#gen.add("angle_increment",double_t, 0, "The angle_increment of the first range measurement [rad].",                     0.00436325,  0.00436325,  0.00436325)
#gen.add("time_increment", double_t, 0, "The time_increment[s].",                                                        0.000046296,   0.000027778,   0.000069444)
gen.add("range_min",      double_t,    0, "The range_min[m].",                                                            0,        0,        30)
gen.add("range_max",      double_t,    0, "The range_max[m].",                                                             30,       0,        30)
gen.add("frequency_scan", int_t,    0, "The mode of frequency.",                                                        1,        1,        2)
gen.add("frame_id",       str_t,    0, "The TF frame in which laser scans will be returned.",                        "laser")


exit(gen.generate(PACKAGE, "wj_716N_lidar", "wj_716N_lidar"))

